cpp/src/io/json/process_tokens.cu

```diff
@@ -87,38 +87,41 @@ void validate_token_stream(device_span<char const> d_input,
 {
   CUDF_FUNC_RANGE();
   if (!options.is_strict_validation()) { return; }
+
+  rmm::device_uvector<bool> d_invalid(tokens.size(), stream);
+  thrust::fill(rmm::exec_policy_nosync(stream), d_invalid.begin(), d_invalid.end(), false);
   using token_t = cudf::io::json::token_t;
-  cudf::detail::optional_trie trie_na =
-    cudf::detail::create_serialized_trie(options.get_na_values(), stream);
-  auto trie_na_view    = cudf::detail::make_trie_view(trie_na);
+  auto literals = options.get_na_values();
+  literals.emplace_back("null");  // added these too to single trie
+  literals.emplace_back("true");
+  literals.emplace_back("false");
+  cudf::detail::optional_trie trie_literals =
```

Comment:
Have you measured any perf impact from the trie use? I never really evaluated its performance.

Reply:
I ran a profile for this and no significant difference.
trie +15.199 ms
direct +15.227 ms